### PR TITLE
Add relative import resolution and Module entity extraction for JS/TS/Python

### PR DESCRIPTION
**Pull request by loganmoon**
## Summary

- Add `current_module_path` parameter to `parse_file_imports()` for resolving relative imports at extraction time
- Add `resolve_relative_import()` for JS/TS (e.g., `./core` → `vanilla.core`)
- Add `resolve_python_relative_import()` for Python relative imports (`.` and `..` syntax)
- Add Module entity handlers for JS/TS/Python to track IMPORTS relationships
- Update all language handlers to derive and pass module_path
- Fix Module handlers to only create entities when imports exist
- Add External node `repository_id` for proper scoping
- Update documentation with new import resolution patterns

This ensures that references stored in entity attributes (`calls`, `uses_types`, `extends_resolved`, `bases_resolved`) match the `qualified_name` format used by entity definitions, enabling proper relationship resolution in Neo4j.

## Test plan

- [x] All 258 library tests pass
- [x] All 47 integration tests pass (JS, Python, TypeScript)
- [ ] Run e2e resolution tests against real codebases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by loganmoon — COMMENTED** [review]
## Code Review Summary

Solid PR that adds important import resolution capabilities for JS/TS/Python. The architecture is well-designed with good separation of concerns. A few issues worth addressing:

### Key Findings

**Performance**: Tree-sitter queries are recompiled on every function call in the utils modules. Consider caching with `OnceLock`.

**Test Coverage**: The new `utils.rs` files for JS and TS have 0% unit test coverage - these functions directly affect relationship creation.

**Documentation**: The onboarding guide incorrectly states that `*_resolved` attributes are used by resolvers - they currently are not.

See inline comments for specific issues.

File: crates/languages/src/javascript/utils.rs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        (call_expression
+          function: (member_expression
+            object: (identifier) @receiver
+            property: (property_identifier) @method))
+    "#;
+
+    let language = tree_sitter_javascript::LANGUAGE.into();
+    let query = match Query::new(&language, query_source) {
+        Ok(q) => q,
+        Err(_) => return Vec::new(),
```
Comment (loganmoon) [inline crates/languages/src/javascript/utils.rs]: Tree-sitter queries are compiled on every call to this function. For files with many functions, this creates significant overhead. Consider caching with `OnceLock`:

```rust
use std::sync::OnceLock;

static JS_CALL_QUERY: OnceLock<Query> = OnceLock::new();

fn get_js_call_query() -> &'static Query {
    JS_CALL_QUERY.get_or_init(|| {
        let language = tree_sitter_javascript::LANGUAGE.into();
        Query::new(&language, JS_CALL_QUERY_SOURCE).expect("Valid query")
    })
}
```

File: crates/languages/src/javascript/utils.rs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            | "undefined"
+            | "symbol"
+            | "bigint"
+            | "never"
+            | "array"
+            | "function"
+            | "promise"
+            | "*"
+    )
+}
```
Comment (loganmoon) [inline crates/languages/src/javascript/utils.rs]: `javascript/utils.rs` has no unit tests. Functions like `extract_function_calls()` and `extract_type_references_from_jsdoc()` directly affect Neo4j relationship creation. Consider adding a test module with coverage for edge cases like nested generics and malformed JSDoc.

File: crates/languages/src/python/utils.rs [hunk 3]
```diff
@@ -202,6 +205,211 @@ pub fn filter_self_parameter(
...
+          function: (attribute
+            object: (identifier) @receiver
+            attribute: (identifier) @method))
+    "#;
+
+    let language = tree_sitter_python::LANGUAGE.into();
+    let query = match Query::new(&language, query_source) {
+        Ok(q) => q,
+        Err(_) => return Vec::new(),
+    };
```
Comment (loganmoon) [inline crates/languages/src/python/utils.rs]: Same tree-sitter query recompilation issue as in `javascript/utils.rs`. The Python `extract_function_calls()` and `extract_type_references()` functions should cache their queries.

File: crates/languages/docs/new-language-onboarding.md [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+                generics: extract_generics(query_match, query, source),
+                is_async: has_modifier(query_match, query, "async"),
+            }),
+        },
+    )?;
+
+    Ok(vec![entity])
+}
+```
+
```
Comment (loganmoon) [inline crates/languages/docs/new-language-onboarding.md]: Documentation mentions `*_resolved` attributes (e.g., `extends_resolved`, `bases_resolved`, `implements_trait_resolved`) are used by resolvers, but the actual resolver implementations only use the base attributes and look up by simple name. Either update the resolvers to use the `_resolved` attributes, or clarify in the docs that these are not currently consumed.

File: crates/e2e-tests/tests/test_resolution_e2e.rs [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+            &neo4j,
+        )
+        .await,
+    );
+
+    // Print final report
+    print_report(&results);
+
+    Ok(())
+}
```
Comment (loganmoon) [inline crates/e2e-tests/tests/test_resolution_e2e.rs]: This e2e test produces metrics but has no assertions on correctness. It validates "something works" but not "it works correctly". Consider adding assertions that verify specific relationships exist (e.g., `assert_relationship_exists(neo4j, repo_id, "UserStore", "CALLS", "fetchUser")`).

**Comment by loganmoon** [comment]
## Merge Blocker: Relationship Resolution Bug

The `*_resolved` attributes (`extends_resolved`, `bases_resolved`, `implements_trait_resolved`) are created but never used. Meanwhile, the resolvers use raw attributes (`extends`, `bases`) with simple name lookup, which will fail when multiple entities share the same name in different modules.

### Current Behavior (Broken)

**Handlers** store both:
- `extends` = `"BaseModel"` (raw name)
- `extends_resolved` = `"myapp.models.base.BaseModel"` (qualified)

**Resolvers** build maps using `name`:
```rust
let class_map = classes.iter().map(|c| (c.name.clone(), c.entity_id.clone()));
```
Then look up the raw attribute: `class_map.get("BaseModel")` - this breaks with name collisions.

### Required Fix

1. **Handlers**: Store the qualified name directly in the base attributes (`extends`, `bases`, `implements_trait`). Remove the `*_resolved` variants entirely.

2. **Resolvers**: Build lookup maps using `qualified_name` instead of `name`:
```rust
let class_map = classes.iter().map(|c| (c.qualified_name.clone(), c.entity_id.clone()));
```

This ensures relationships resolve correctly even when multiple entities share the same simple name.

**Comment by loganmoon** [comment]
### Inline Feedback

**`crates/outbox-processor/src/neo4j_relationship_resolver.rs:487-529` - Duplicate `resolve_import_path`**  
This duplicates `resolve_relative_import` from `import_map.rs`. The extension stripping logic here differs from `import_map.rs` which only strips known JS/TS extensions. Consider extracting to a shared utility.

**`crates/outbox-processor/src/neo4j_relationship_resolver.rs:276-278, 384, 450, 567` - Silent JSON failures**  
`unwrap_or_default()` on JSON deserialization silently swallows malformed data. Consider logging warnings to aid debugging.

**`crates/outbox-processor/src/neo4j_relationship_resolver.rs:440-444` - `CallGraphResolver` lookup collision risk**  
Using both `name` and `qualified_name` as lookup keys could cause collisions where a simple name shadows a qualified name from a different module. Other resolvers use only `qualified_name`.

**`crates/outbox-processor/src/neo4j_relationship_resolver.rs:624, 713` - Duplicate `get_all_entities()` calls**  
`get_all_entities()` is called in both `ContainsResolver::resolve()` and `resolve_external_references()`, loading all entities with full `content` twice. Consider caching at the `resolve_pending_relationships()` level.

**`crates/languages/src/typescript/handler_impls/module_handlers.rs:137-143` - Inconsistent Module creation**  
TypeScript always creates a Module entity, but JS/Python return early when no imports. This behavioral difference should be tested or made consistent.
